src/main/java/com/yahoo/bullet/querying/Querier.java

```diff
@@ -706,4 +714,18 @@ private void incrementRate() {
     private String getMetaKey() {
         return metaKeys.getOrDefault(Meta.Concept.QUERY_METADATA.getName(), null);
     }
+
+    private void addToTransientFields(PostStrategy postStrategy) {
+        Projection projection = runningQuery.getQuery().getProjection();
+        Aggregation aggregation = runningQuery.getQuery().getAggregation();
+        if (aggregation.getType() == Aggregation.Type.RAW || projection != null) {
+            Map<String, String> projectionFields = projection.getFields();
+            List<String> requiredFields = postStrategy.getRequiredFields();
+            for (String field : requiredFields) {
+                if (!projectionFields.containsKey(field) || !projectionFields.containsValue(field)) {
+                    transientFields.put(field, field);
```

Comment:
What happens if there's a projection a->b, and there's a required field "a" ?

Reply:
I guess it will add a transient projection a->a and delete "a" at end, otherwise the result record has no field named "a".